adm/src/commands/batch.rs

```diff
@@ -0,0 +1,178 @@
+        }
+    } else if format == "json" {
+        for batch in batches {
+            println!(
+                "{}",
+                serde_json::to_string_pretty(&JsonYamlBatch::from(&batch.unwrap())).map_err(
+                    |err| CliError::EnvironmentError(format!(
+                        "Cannot format batches into json: {}",
+                        err
+                    ))
+                )?
+            )
+        }
+    } else if format == "yaml" {
+        for batch in batches {
+            println!(
+                "{}",
+                serde_yaml::to_string(&JsonYamlBatch::from(&batch.unwrap())).map_err(|err| {
+                    CliError::EnvironmentError(format!("Cannot format batches into yaml: {}", err))
+                })?
+            )
+        }
+    } else {
+        let data = parse_batches(batches)?;
+        print_table(data);
+    }
+    Ok(())
+}
+
+fn parse_batches(
+    batches: Box<dyn Iterator<Item = Result<Batch, ClientError>>>,
+) -> Result<Vec<Vec<String>>, CliError> {
+    let mut data = Vec::new();
+    data.push(vec![
+        "BATCH_ID".to_string(),
+        "TXNS".to_string(),
+        "SIGNER".to_string(),
+    ]);
+
+    for batch in batches {
+        let batch =
+            batch.map_err(|err| CliError::ParseError(format!("Failed to get batch: {}", err)))?;
+        data.push(vec![
+            batch.header_signature.to_string(),
+            batch.transactions.len().to_string(),
+            batch.header.signer_public_key.to_string(),
+        ]);
+    }
+    Ok(data)
+}
+
+/// Attempt to parse the given credentials formatted as "username:password"
+fn get_credentials(auth: &str) -> Result<Vec<&str>, CliError> {
+    match auth.splitn(2, ':').collect::<Vec<&str>>() {
+        credentials if credentials.len() == 2 => Ok(credentials),
+        _ => Err(CliError::ArgumentError(
+            "Username and password formatted incorrectly".to_string(),
+        )),
+    }
+}
+
+fn print_table(table: Vec<Vec<String>>) {
+    let mut max_lengths = Vec::new();
+
+    // find the max lengths of the columns
+    for row in table.iter() {
+        for (i, col) in row.iter().enumerate() {
+            if let Some(length) = max_lengths.get_mut(i) {
+                if col.len() > *length {
+                    *length = col.len()
+                }
+            } else {
+                max_lengths.push(col.len())
+            }
+        }
+    }
+
+    // print each row with correct column size
+    for row in table.iter() {
+        let mut col_string = String::from("");
+        for (i, len) in max_lengths.iter().enumerate() {
+            if let Some(value) = row.get(i) {
+                col_string += &format!("{}{} ", value, " ".repeat(*len - value.len()),);
+            } else {
+                col_string += &" ".repeat(*len);
+            }
+        }
+        println!("{}", col_string);
+    }
+}
+
+#[derive(Debug, Serialize)]
+struct JsonYamlBatch {
```

Comment:
Maybe this could be called StructuredOutputBatch

Reply:
Fixed